### ISSUE TITLE
Feedback messages to users when a job is failing

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#172]
In GitLab by @jordicuadrado on Jul 29, 2016, 13:06

Hi AS team!

During the auto-nmmb testing period, @kserradell started the conversation on the opportunity to change the output messages that an user receives when a job is failing.

To reproduce the case, a bash script which has some basic checks on configuration parameters introduced by an user (retrieved form .conf files) executes an exit code when an error is detected:
```bash
exit 1
```
(thus making the JOB to Fail)  
(Example: https://earth.bsc.es/gitlab/es/auto-nmmb/blob/production/templates/00_local_send_initial.sh#L131)

Autosubmit after a short period of time will correctly change the status of that job to FAILED. However, in the meantime, the following output is produced:

```bash
40 of 40 jobs remaining (12:34)
nohup: ignoring input and redirecting stderr to stdout
cp: cannot stat ‘/home/Earth/jcuadrad/autosubmit/a000/tmp/LOG_a000/a000_20000529_fc0_1_LOCAL_SEND_INITIAL_COMPLETED’: No such file or directory
cp: cannot stat ‘/home/Earth/jcuadrad/autosubmit/a000/tmp/LOG_a000/a000_20000529_fc0_1_LOCAL_SEND_INITIAL_COMPLETED’: No such file or directory
cp: cannot stat ‘/home/Earth/jcuadrad/autosubmit/a000/tmp/LOG_a000/a000_20000529_fc0_1_LOCAL_SEND_INITIAL_COMPLETED’: No such file or directory
cp: cannot stat ‘/home/Earth/jcuadrad/autosubmit/a000/tmp/LOG_a000/a000_20000529_fc0_1_LOCAL_SEND_INITIAL_COMPLETED’: No such file or directory
cp: cannot stat ‘/home/Earth/jcuadrad/autosubmit/a000/tmp/LOG_a000/a000_20000529_fc0_1_LOCAL_SEND_INITIAL_COMPLETED’: No such file or directory
cp: cannot stat ‘/home/Earth/jcuadrad/autosubmit/a000/tmp/LOG_a000/a000_20000529_fc0_1_LOCAL_SEND_INITIAL_COMPLETED’: No such file or directory
[WARNING] Job a000_20000529_fc0_1_LOCAL_SEND_INITIAL seemed to be completed but there is no COMPLETED file
[WARNING] Job a000_20000529_fc0_1_LOCAL_SEND_INITIAL in UNKNOWN status
```

This information could be a bit confusing to the user. (Although is an earlier indicator that the execution will fail).
  
A suggestion is to change the message from:  
*[WARNING] Job (...) seemed to be completed but there is no COMPLETED file*  
to:  
*(...) seemed to be* **finished** *but there is no COMPLETED file*  
and hiding the cp messages?  
  
It's just a suggestion/an approach to clarify the feedback that newer users of auto-models receive.

Many thanks!!

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Jul 29, 2016, 14:36

Hi @jordicuadrado,

Thanks for the suggestions!

One the one hand, the 'cp' logs (or similar) has already been removed from some platforms (available since next version). However we will review all the platforms in order to avoid this kind of useless outputs.

On the other hand, the **completed** message is a bit complex. In general terms, Autosubmit only will check if there's a _COMPLETED_ file when the scheduler returns that the job has finished correctly. So almost always it's OK. The problem is that the local platform does not has an scheduler that can say how has finished the job. In this particular case, each finished job is treated as completed (the reason why you are seeing this message).

In conclusion, I think that in general terms, this message is quite clear, and in my opinion, doesn't make sense changing this message to make clearer a specific case.

**Comment by kinow** [comment]
In GitLab by @ jlope2 on Oct 13, 2016, 23:35

Milestone removed

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 4, 2018, 15:16

Since this is quite old and is about clarity, I would like to know if the answer by @ jlope2 was satisfactory.

Also, due the changes in clarity during 2 years, I think that is better to close this issue, and open another with a similar suggestion or related to this one. So for now I am closing this.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 4, 2018, 15:16

closed
